lib/optimizely.rb

```diff
@@ -201,24 +201,14 @@ def track(event_key, user_id, attributes = nil, event_tags = nil)
 
       return nil unless user_inputs_valid?(attributes, event_tags)
 
-      experiment_ids = @config.get_experiment_ids_for_event(event_key)
-      if experiment_ids.empty?
-        @config.logger.log(Logger::INFO, "Not tracking user '#{user_id}'.")
+      event = @config.get_event_from_key(event_key)
+      unless event
+        @config.logger.log(Logger::INFO, "Not tracking user '#{user_id}' for event '#{event_key}'.")
         return nil
       end
 
-      # Filter out experiments that are not running or that do not include the user in audience conditions
-
-      experiment_variation_map = get_valid_experiments_for_event(event_key, user_id, attributes)
-
-      # Don't track events without valid experiments attached
-      if experiment_variation_map.empty?
-        @logger.log(Logger::INFO, "There are no valid experiments for event '#{event_key}' to track.")
-        return nil
-      end
-
-      conversion_event = @event_builder.create_conversion_event(event_key, user_id, attributes,
-                                                                event_tags, experiment_variation_map)
+      conversion_event = @event_builder.create_conversion_event(event_key, user_id, attributes, event_tags)
```

Comment:
May be update `create_conversion_event` to take `event` instead of sending `event_key` and retrieving it every time.